### PR TITLE
Fake package cleaning

### PR DESCRIPTION
**Pull request by Gilthoniel**
This cleans the fake error to use a random value every time the unit tests are run so that it prevents hardcoded values. It also cleans the "oops" errors to use the fake one.

### CODE REVIEW

**Comment by Gilthoniel** [comment]
This will probably conflict with previous PRs so it should not be merged right away but after the others.

File: internal/testing/fake/crypto.go [hunk 1]
```diff
@@ -0,0 +1,388 @@
+package fake
+
+import (
+	"hash"
+
+	"go.dedis.ch/dela/crypto"
+	"go.dedis.ch/dela/serde"
+)
+
+// PublicKeyFactory is a fake implementation of a public key factory.
```
Comment (nkcr) [inline internal/testing/fake/crypto.go:10]: In an ideal world we would keep the `- implements ...` convention, same for the functions. What do you think ?

Comment (Gilthoniel) [reply internal/testing/fake/crypto.go:10]: Yes, you're right, I wanted to do it but forgot to update the comments.

**Review by nkcr — APPROVED** [review]
I'm gonna miss the `xerrors.New("oops")` 😢

**Pull request merged**